### PR TITLE
Assignment 5 - Timothy Nelson

### PR DESCRIPTION
**Pull request by timmytaters**
@timmytaters
https://red-bush-0be0c391e.5.azurestaticapps.net/

I know this is missing a lot, I was short on time between having to learn Azure, (Which is why a lot of the past work won't load) my senior project, and also starting work on the final.

### CODE REVIEW

**Comment by timmytaters** [comment]
Overall the word editor is working, but the login part is not, and the common words toggle is a WIP

**Review by Hex-Paladin — COMMENTED** [review]
Sorry for the late review, I have been having issues catching up with my senior project and other responsibilities too,

Create an editor for the available words ✅
Create a page named "WordEditor" that allows words to be edited ✅
Support deleting a word ❌✅
Support adding a new word ❌✅
Editing words is not supported ✅
Duplicate words should not be allowed ✅
Add & support changing the common word boolean flag ❌✅
Filter the list of returned words by common words ❌
Words should be sorted alphabetically ✅
Provide a text box for searching words AS YOU TYPE which supports searching for words starting with letters specified ✅
Provide a menu item for navigating to the WordEditor page ✅
Anyone can look at the word list ✅
Word list is paginated on the server side (10-100 per page) ✅
Any logged in user can change the common word flag ❌
There is a login page/component/popup/etc for users to sign in ✅
Only users over 21 years of age (based on birthday) with a claim of MasterOfTheUniverse can add and remove words ❌✅
Users can be set up at start up in the API project as a seed (apriori), there does not need to be a sign up mechanism ❌✅

I was unable to login or add/delete words so it was hard to test some of the requirements.

**Comment by timmytaters** [comment]
The word add/delete should be working, it just doesn't always automatically refresh. If you type a bit in the search it should pop up with/without the word

File: wordle-app/pages/editor.vue [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  function validWords(): string[] {
+    return words;
+  }
+
+  function removeWord(word:string){
+    const index = words.indexOf(word, 0);
+    if (index > -1) {
+        words.splice(index, 1);
+    }
+  }
```
Comment (A-Tanner) [inline wordle-app/pages/editor.vue:85]: The logic here, and in addNewWord, only affects the local list of words. I don't see any logic that propagates these changes to the API, which was the ask.

**Review by A-Tanner — COMMENTED** [review]
Create an editor for the available words ✅
Create a page named "WordEditor" that allows words to be edited ✅
Support deleting a word ❌ -- Site appears to crash. Deletion does not persist (local only).
Support adding a new word ❌ -- Site appears to crash. Addition does not persist (local only).
Editing words is not supported ✅
Duplicate words should not be allowed ✅ I see there is a check for this in the frontend code.
Add & support changing the common word boolean flag ❌
Filter the list of returned words by common words ✅-- Filter appears to exist, but crashes the site
Words should be sorted alphabetically ✅
Provide a text box for searching words AS YOU TYPE which supports searching for words starting with letters specified ❌
Provide a menu item for navigating to the WordEditor page ✅
Anyone can look at the word list ✅
Word list is paginated on the server side (10-100 per page) ✅
Any logged in user can change the common word flag ❌ -- Flag is not supported
There is a login page/component/popup/etc for users to sign in ❌ -- Popup is there, but there is no sign in support
Only users over 21 years of age (based on birthday) with a claim of MasterOfTheUniverse can add and remove words ❌ -- Code shows no restrictions on add, remove
Users can be set up at start up in the API project as a seed (apriori), there does not need to be a sign up mechanism ❌ -- Unsure, but login isn't supported

Tim, thank you for being upfront about your difficulties with this assignment. I hope you can relax for the Summer now that Assignment 5 is turned in. Best of luck in the future!

**Pull request closed without merging**